cardano-db-sync/src/Cardano/DbSync/Plugin/Default/InsertShelley.hs

```diff
@@ -0,0 +1,131 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Cardano.DbSync.Plugin.Default.InsertShelley
+  ( insertShelleyBlock
+  ) where
+
+import           Cardano.Prelude
+
+import           Cardano.BM.Trace (Trace, logDebug, logInfo)
+
+import           Control.Monad.Logger (LoggingT)
+import           Control.Monad.Trans.Except.Extra (runExceptT)
+
+import qualified Cardano.Crypto as Crypto
+
+import           Control.Monad.IO.Class (liftIO)
+import           Control.Monad.Trans.Reader (ReaderT)
+
+import qualified Data.ByteString.Char8 as BS
+import qualified Data.ByteString.Base16 as Base16
+import qualified Data.Text.Encoding as Text
+
+import           Database.Persist.Sql (SqlBackend)
+
+import qualified Cardano.Db as DB
+import           Cardano.DbSync.Error
+import           Cardano.DbSync.Util (textShow, renderByteArray)
+
+import qualified Shelley.Spec.Ledger.BlockChain as SL
+import           Shelley.Spec.Ledger.Tx as SL
+
+import           Ouroboros.Consensus.Shelley.Ledger (ShelleyBlock (..), Crypto)
+import           Ouroboros.Network.Block (BlockNo (..), Tip, SlotNo (..), getTipBlockNo)
+import           Ouroboros.Network.Point (withOrigin)
+
+
+mkSlotLeader :: Crypto c => SL.Block c -> DB.SlotLeader
+mkSlotLeader blk =
```

Comment:
Since that is an internal function that is not exported, can we have that at the bottom of the file?

Reply:
Sure.